main/src/ca/uwaterloo/flix/language/phase/Weeder2.scala

```diff
@@ -1608,7 +1608,7 @@ object Weeder2 {
       mapN(pickExpr(tree)) {
         expr =>
           idents.foldLeft(expr) {
-            case (acc, ident) => Expr.RecordSelect(acc, Name.mkLabel(ident), ident.loc)
+            case (acc, ident) => Expr.RecordSelect(acc, Name.mkLabel(ident), tree.loc)
```

Comment:
does this fix work for `r#x#y#z`?

Reply:
Yeah, you're right. I think I've fixed it. We want the location that from tree.loc through ident.loc for each

Reply:
I believe I've fixed it now. I've tested it on a nested example (similar to what you proposed. Good catch. Before they all had the same sourcelocation, which ofc. makes no sense.